app/src/main/java/ca/marshallasch/veil/HashTableStore.java

```diff
@@ -103,17 +110,18 @@ public static synchronized HashTableStore getInstance(@Nullable final Context c)
     }
 
     /**
-     * This function will update the saved copy of the hash map to the disk.
+     * This function will update the saved copy of the hash map to the disk. The file needs to be
+     * created in the constructor if a non null context was passed to it. Otherwise the table is not
+     * saved.
      */
-    public void save(@Nullable Context context) {
+    public void save() {
 
-        // if there is no context then don't save the file
-        if (context == null) {
+        // if the file does not exist then do not save, or if it has not been modified
+        if (mapFile == null || !modified) {
+            Log.d("HashTableStore", "Not actually saving the file, changed: " + String.valueOf(modified));
             return;
         }
 
```

Comment:
Why was this `File mapFile = new File(context.getFilesDir(), "HASH_MAP");` deleted. Its later called in the try block below.

Reply:
The file is now created in the constructor instead of in the save function